### DISCUSSION TITLE
A proposal for Rework for Tech difficulty

### DISCUSSION THREAD

**Discussion by akivn** [ppy/osu#30492] [Ideas]
Moved from the reply in https://github.com/ppy/osu/discussions/16122#discussioncomment-11079866

Alt and Tech maps are being underweight and have been causing some issues with players thinking the map was easy only to be freaked out by those sliders and sudden jumps. Therefore, I would suggest building a new difficulty factor called "Skill", addressed solely for technical maps (alt/tech).

My idea of the "Skill" calculation will be based on these factors:

1. Flow and Aim variation

This is based on how much the aim angle and real aim (jump-distance)/lazy aim changes throughout the map.

Situations like the first image shown on the original post (https://osu.ppy.sh/beatmapsets/1357624#osu/2809623) where there is big variation in short succession will be massively buffed due to this.

This factor will be based on how much the "angle value" changes and how fast these jumps are, and also how much variation of lazy path and jump distance changes. For the variation of jump distances, a "jumpVariation" pattern can be set up, where it is based on a weighted average of lazy aim and real aim distances.
Super-small jumps with a higher degree of overlaps will be valued less, slowly but exponentially increasing from a very low value in completely overlapped cases to 1x in completely separated cases. The multiplier calculation for both of the jumps will be calculated as well. Stacks will also be calculated as completely overlapped cases just like how the current pp system (and also the new rework) works.

This factor is also scaled by how fast the jumps are. Slower jumps at 150-200ms (150-200bpm) gets less valuable, while this scales up as jumps become faster, massively increasing after a certain value (e.g. 90-100ms) where the alt genre gradually turns into tech.

e.g. https://osu.ppy.sh/beatmapsets/2074198#osu/4341639 - pattern buffed significantly
image
The pattern is a constant change of angles. It also consists of some variation of spacings (as well as lazy aim), so despite some jumps are being nerfed by small spacing the high jump density (185bpm 1/4s) and large variation inside a single strain gives a very high multiplier.

2. Slider Velocity and Length

Kicksliders (which is lazy aim = 0) has a difficulty of zero and does not make a difference in terms of SR (or even lower), yet tech is abusing them to make different difficulty spikes and intensity parts.

This factor will account the real slider movement and the lazy aim (weighted, with lazy aim > slider movement) - once the slider has a length over a certain value there will be a buff towards the "Skill" component. If the lazy aim component ≠ 0, or the slider has high slider velocity, the buff will be much more significant. This might also fix slider jumps in generic maps giving less pp and lower SR than normal jumps.

e.g. https://osu.ppy.sh/beatmapsets/1889729#osu/3912664
Maps with long and fast sliders (mostly alternating / tech) with be buffed by this component as well, such as this featuring generic alternating patterns and sliders.

The above two factors should cover most, if not all, alt / tech maps.

The "Skill" component can be separate from the current "Aim" / "Speed" / "Strain" (which fulfills the idea of the original suggestion) and the SR from skill can be directly addictive towards the final SR (and nerfing the base multiplier for the other two components to balance out). This keeps the current component mostly untouched which is much easier to work on with.

If a new component is not possible, It might be possible to implement the above factors to aim as well (by implementing new bonuses).

**Comment by TextAdventurer12** [comment]
couldn't angle variation just be inside of aim and slider velocity inside of aim with sliders? I don't see the point in creating a new skill (and 'Skill' seems like a poor name for it in any case)

**Reply by akivn** [reply]
Yea, I just put them into strain for angle deviation calculation
![image](https://github.com/user-attachments/assets/a59143c1-3cbd-41c5-a7dd-c86f51cbb0bf)

**Comment by akivn** [comment]
why is it in taiko not std lmao
